influxql/ast_test.go

```diff
@@ -1700,6 +1701,102 @@ func TestParse_Errors(t *testing.T) {
 	}
 }
 
+// This test checks to ensure that we have given thought to the database
+// context required for security checks.  If a new statement is added, this
+// test will fail until it is categorized into the correct bucket below.
+func Test_EnforceHasDefaultDatabase(t *testing.T) {
+	pkg, err := importer.Default().Import("github.com/influxdata/influxdb/influxql")
+	if err != nil {
+		fmt.Printf("error: %s\n", err.Error())
+		return
+	}
+	statements := []string{}
+
+	// this is a list of statements that do not have a database context
+	exemptStatements := []string{
+		"CreateDatabaseStatement",
+		"CreateUserStatement",
+		"DeleteSeriesStatement",
+		"DropDatabaseStatement",
+		"DropMeasurementStatement",
+		"DropSeriesStatement",
+		"DropShardStatement",
+		"DropUserStatement",
+		"GrantAdminStatement",
+		"KillQueryStatement",
+		"RevokeAdminStatement",
+		"SelectStatement",
+		"SetPasswordUserStatement",
+		"ShowContinuousQueriesStatement",
+		"ShowDatabasesStatement",
+		"ShowDiagnosticsStatement",
+		"ShowGrantsForUserStatement",
+		"ShowQueriesStatement",
+		"ShowRetentionPoliciesStatement",
+		"ShowSeriesStatement",
+		"ShowShardGroupsStatement",
+		"ShowShardsStatement",
+		"ShowStatsStatement",
+		"ShowSubscriptionsStatement",
+		"ShowTagKeysStatement",
+		"ShowTagValuesStatement",
```

Comment:
And here https://github.com/influxdata/influxdb/blob/9e6c9899399442ac61c5d9a6be61c3c1548a0e4d/influxql/ast.go#L3225